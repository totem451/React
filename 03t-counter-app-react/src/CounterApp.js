import React, {useState} from 'react';
import PropTypes from 'prop-types';

const CounterApp = ( { value = 100 } ) => {

    // const state = useState('Goku');
    // console.log(state);

    // const [nombre, setNombre] = useState('Goku');
    // console.log(nombre, setNombre);

    const [counter, setCounter] = useState( value ); //[]

    //handleAdd
    const handleAdd = (/*e*/) => {

        // console.log(e)

        // return () => console.log('+1');

        setCounter( counter + 1 );

        // setCounter( (counter) => counter + 1 )
    }

    const handleSubstract = () => {

        setCounter( counter - 1 );

    }

    const handleReset = () => setCounter( value );

    return (
        <>
            <h1>Counter App</h1>
            {/* <h2> { value } </h2> */}
            <h2> { counter } </h2>
            <button onClick = { /*handleAdd()*/ handleAdd /*() => { console.log('+1') }*/ }>+1</button>
            <button onClick = { handleReset }>reset</button>
            <button onClick = { handleSubstract }>-1</button>
        </>
    );

}

CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp;