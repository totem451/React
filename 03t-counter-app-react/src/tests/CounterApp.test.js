import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';
import React from 'react';

describe('Pruebas en CounterApp ', () => {

    const valueg = 420;

    // let wrapper;

    let wrapper = shallow(<CounterApp value = {valueg}/>);

    beforeEach(() => {
        wrapper = shallow(
            <CounterApp 
                value = {valueg}
            />
        );

    })
    
    test('debe mostrar <CounterApp /> correctamente ', () => {

        const wrapper = shallow(
            <CounterApp />
        );

        expect(wrapper).toMatchSnapshot();
    })

    test('debe mostrar el valor por defecto de 100 ', () => {
        
        const value = 100;

        const wrapper = shallow(
            <CounterApp 
                value = {value}
            />
        );

        const textoH2 = wrapper.find('h2').text().trim();

        expect(textoH2).toBe(value.toString());

    })

    test('debe incrementar con el boton +1 ', () => {

        // const btn1 = wrapper.find('button').at(0).simulate('click');
        // const btn2 = wrapper.find('button').at(1);
        // const btn3 = wrapper.find('button').at(2);

        wrapper.find('button').at(0).simulate('click');

        // console.log(btn1.html());

        const textoH2 = wrapper.find('h2').text().trim();

        expect(textoH2).toBe((valueg + 1).toString());

    })

    test('debe decrementar con el boton -1  ', () => {

        wrapper.find('button').at(2).simulate('click');

        const textoh2 = wrapper.find('h2').text().trim();

        expect(textoh2).toBe((valueg - 1).toString());

    })

    test('debe resetear con el boton reset  ', () => {
        
        wrapper.find('button').at(0).simulate('click');

        console.log(wrapper.find('h2').text().trim());

        wrapper.find('button').at(1).simulate('click');

        console.log(wrapper.find('h2').text().trim());

        const textoh2 = wrapper.find('h2').text().trim();

        expect(textoh2).toBe((valueg).toString());

    })
    
})
