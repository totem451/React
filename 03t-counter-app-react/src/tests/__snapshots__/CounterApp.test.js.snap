// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp  debe mostrar <CounterApp /> correctamente  1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
     
    100
     
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
